samples/algorithms/integer-factorization/FlameGraphResourcesEstimator.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+using Microsoft.Quantum.Simulation.QCTraceSimulatorRuntime;
+using Microsoft.Quantum.Simulation.Simulators.QCTraceSimulators;
+
+namespace Microsoft.Quantum.Simulation.Simulators {
+    public class FlameGraphResourcesEstimator : ResourcesEstimator {
+        private FlameGraphCounter operationsCounter;
+        public static new QCTraceSimulatorConfiguration RecommendedConfig() =>
+            new QCTraceSimulatorConfiguration
+            {
+                CallStackDepthLimit = Int32.MaxValue,
+
+                ThrowOnUnconstrainedMeasurement = false,
+                UseDistinctInputsChecker = false,
+                UseInvalidatedQubitsUseChecker = false,
+
+                UsePrimitiveOperationsCounter = false,
+                UseDepthCounter = true,
+                UseWidthCounter = true
+            };
+
+        public FlameGraphResourcesEstimator() : this(RecommendedConfig()) {}
+        public FlameGraphResourcesEstimator(QCTraceSimulatorConfiguration config, PrimitiveOperationsGroups resourceToVisualize = PrimitiveOperationsGroups.CNOT) : base(WithoutPrimitiveOperationsCounter(config)) {
+            operationsCounter.ResourceToVisualize = resourceToVisualize;
+        }
+
+        private static QCTraceSimulatorConfiguration WithoutPrimitiveOperationsCounter(QCTraceSimulatorConfiguration config) {
+            config.UsePrimitiveOperationsCounter = false;
+            return config;
+        }
+ 
+        protected override void InitializeQCTracerCoreListeners(IList<IQCTraceSimulatorListener> listeners) {
```

Comment:
Yes, it's `IList` in [here](https://github.com/microsoft/qsharp-runtime/blob/df59efb7e11fc62439c777a85d0da95db4604d68/src/Simulation/Simulators/QCTraceSimulator/QCTraceSimulatorImpl.cs#L118), and I think it needs to be, since it can be modified.